packages/demo-cards/claim.gts

```diff
@@ -0,0 +1,127 @@
+import { Chain } from './chain';
+import {
+  Card,
+  contains,
+  field,
+  StringCard,
+  linksTo,
+  Component,
+} from 'https://cardstack.com/base/card-api';
+import { Button, CardContainer, FieldContainer } from '@cardstack/boxel-ui';
+import BooleanCard from 'https://cardstack.com/base/boolean';
+
+declare global {
+  interface Window {
+    ethereum: any;
+  }
+}
+
+export class Claim extends Card {
+  static displayName = 'Claim';
+  @field
+  moduleAddress = contains(StringCard);
+  @field
+  safeAddress = contains(StringCard);
+  @field
+  explanation = contains(StringCard);
+  @field
+  signature = contains(StringCard);
+  @field
+  encoding = contains(StringCard);
+  @field
+  chain = linksTo(() => Chain);
+  @field title = contains(StringCard, {
+    computeVia: function (this: Claim) {
+      return `Claim for ${this.safeAddress}`;
+    },
+  });
+  @field connected = contains(BooleanCard, {
```

Comment:
hmmm, i don't think this is a good use of a field. This is transitory application session state. A field is meant to be persistent state for a card.

Reply:
Ok Im not really sure that this was the right approach. But, I couldn't just access @model.connected within the isolated template since `get connected()` would still return a Promise<boolean>. The reason why Promise is because the I need to access metamask api which is async. So I resorted to adding this to isolated. The obvious downside is tha glint seems to complain "Decorators are not valid here" and I have to import `import { tracked } from '@glimmer/tracking';` (not shimmed I haven't dealt with that). But it works ok on the browser. 

```
    @tracked connected: any;
    constructor(owner: unknown, args: any) {
      super(owner, args);
      this.initialize();
    }

    async initialize() {
      this.connected = await this.args.model.connected;
    }
```

Reply:
I think I probably need to spend time understanding the constraints put forth on the Card abstraction